src/Middleware/SpaServices.Extensions/src/Npm/NpmMiddleware.cs

```diff
@@ -0,0 +1,136 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Builder;
+using Microsoft.Extensions.Logging;
+using Microsoft.AspNetCore.NodeServices.Npm;
+using Microsoft.AspNetCore.NodeServices.Util;
+using Microsoft.AspNetCore.SpaServices.Util;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using System.Threading;
+using System.Net.Http;
+using Microsoft.AspNetCore.SpaServices.Extensions.Util;
+
+namespace Microsoft.AspNetCore.SpaServices.Npm
+{
+    public static class NpmMiddleware
+    {
+        private const string LogCategoryName = "Microsoft.AspNetCore.SpaServices";
+
+        public static void Attach(ISpaBuilder spaBuilder,
+            string npmScriptName,
+            Func<int, string> npmParameters,
+            Func<int, IDictionary<string, string>> envVars, 
+            Regex outputMatch, Func<Match, int, Uri> listeningUri)
+        {
+            var sourcePath = spaBuilder.Options.SourcePath;
+            if (string.IsNullOrEmpty(sourcePath))
+            {
+                throw new ArgumentException("Cannot be null or empty", nameof(sourcePath));
+            }
+
+            if (string.IsNullOrEmpty(npmScriptName))
+            {
+                throw new ArgumentException("Cannot be null or empty", nameof(npmScriptName));
+            }
+
+            // Start Angular CLI and attach to middleware pipeline
```

Comment:
Comment is odd, as this is not specific to Angular

Reply:
Fixed